src/bundle/Resources/public/js/alloyeditor/src/buttons/ez-btn-linkedit.js

```diff
@@ -0,0 +1,359 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AlloyEditor from 'alloyeditor';
+
+export default class EzBtnLinkEdit extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = this.getInitialState()
+    }
+
+    static get key() {
+        return 'ezlinkedit';
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState(this.getInitialState());
+    }
+
+    componentWillUnmount() {
+        if (!this.state.discoveringContent && this.state.isTemporary) {
+            this.removeLink();
+        }
+    }
+
+    /**
+     * Lifecycle. Invoked once before the component is mounted.
+     * The return value will be used as the initial value of this.state.
+     *
+     * @method getInitialState
+     */
+    getInitialState() {
+        const linkUtils = new CKEDITOR.Link(this.props.editor.get('nativeEditor'), {appendProtocol: false});
+        let link = linkUtils.getFromSelection();
+        let href = '';
+        let target = '';
+        let title = ''
+        let isTemporary = false;
+
+        if (link) {
+            href = link.getAttribute('href');
+            target = link.hasAttribute('target') ? link.getAttribute('target') : target;
+            title = link.getAttribute('title');
+            isTemporary = link.hasAttribute('data-ez-temporary-link');
+        } else {
+            linkUtils.create(href, {"data-ez-temporary-link": true});
```

Comment:
Is the result of the `.create()` function call assigned to `linkUtils`?
I'm not sure I understand this behaviour correctly.

Reply:
`linkUtils` is an instance of `CKEditor.Link`. In this line, it creates a link and in line below it's assigned to the `link`